packages/zoe/src/contracts/newSwap/getCurrentPrice.js

```diff
@@ -52,16 +52,22 @@ export const makeGetCurrentPrice = (
       // remainder to the pool to get a quote. Then we'll add the fee to deltaX
       // before sending out the quote.
 
-      // prelimProtocolFee will be replaced by protocolFee once we get the final
-      // value of amountIn from getPriceGivenAvailableInput.
-      const prelimProtocolFee = multiplyBy(amountIn, protocolFeeRatio);
-      const poolAmountIn = amountMath.subtract(amountIn, prelimProtocolFee);
+      // amountIn will be divided into deltaX (what's added to the pool) and the
+      // protocol fee. protocolFee will be protocolFeeRatio * deltaX.
+      // Therefore, amountIn = (1 + protocolFeeRatio) * deltaX, and
+      // protocolFee =  protocolFeeRatio * amountIn / (1 + protocolFeeRatio).
+      const feeOverOnePlusFee = makeRatio(
+        protocolFeeBP,
+        centralBrand,
+        add(BASIS_POINTS, protocolFeeBP),
+      );
+      const protocolFee = multiplyBy(amountIn, feeOverOnePlusFee);
```

Comment:
Ah, good point. I think the confusing language is `const poolAmountIn = AmountMath.subtract(amountIn, protocolFee);`

Can we call it `deltaX` instead of `poolAmountIn`? Maybe add a comment explaining that both fees are charged from `deltaX`?

Reply:
Oh, I see you added that comment already. Thanks!